packages/nx/src/utils/assert-workspace-validity.ts

```diff
@@ -19,44 +16,16 @@ export function assertWorkspaceValidity(
 
   const invalidImplicitDependencies = new Map<string, string[]>();
 
-  Object.entries<'*' | string[] | ImplicitJsonSubsetDependency>(
-    nxJson.implicitDependencies || {}
-  )
-    .reduce((acc, entry) => {
-      function recur(value, acc = [], path: string[]) {
-        if (value === '*') {
-          // do nothing since '*' is calculated and always valid.
-        } else if (typeof value === 'string') {
-          // This is invalid because the only valid string is '*'
-          throw new Error(stripIndents`
-         Configuration Error 
-         nx.json is not configured properly. "${path.join(
-           ' > '
-         )}" is improperly configured to implicitly depend on "${value}" but should be an array of project names or "*".
-          `);
-        } else if (Array.isArray(value)) {
-          acc.push([entry[0], value]);
-        } else {
-          Object.entries(value).forEach(([k, v]) => {
-            recur(v, acc, [...path, k]);
-          });
-        }
-      }
-
-      recur(entry[1], acc, [entry[0]]);
-      return acc;
-    }, [])
-    .reduce((map, [filename, projectNames]: [string, string[]]) => {
-      detectAndSetInvalidProjectGlobValues(
-        map,
-        filename,
-        projectNames,
-        projects,
-        projectNames,
-        projectNameSet
-      );
-      return map;
-    }, invalidImplicitDependencies);
+  if (nxJson.implicitDependencies) {
+    output.warn({
+      title:
+        'Using `implicitDependencies` for global implicit dependencies configuration is no longer supported.',
+      bodyLines: [
+        'Use "namedInputs" instead.',
+        'For more information about the usage of "namedInputs" see https://nx.dev/deprecated/global-implicit-dependencies#global-implicit-dependencies',
```

Comment:
Can you include a line to use `nx repair`? But I noticed that `nx repair` will mess up the `production` fileset. Can you fix that as well please?

Reply:
Sure, fix is in a separate PR: https://github.com/nrwl/nx/pull/16145